### PR TITLE
Selection for GeoDataFrame `engine` in plotting routines 

### LINKED ISSUE #988: Select `engine` for GeoDataFrame when plotting
Add an option in plotting routine that internally construct a `GeoDataFrame` to select an engine. Either `spatialpandas` or `geopandas`

### PR DESCRIPTION
**Pull request by philipc2**
Closes #988

## Overview

* Adds support for selecting which GeoDataFrame engine to use when plotting
* Fixes internal central coordinate and projections leftover from #922

### CODE REVIEW

File: uxarray/grid/grid.py [hunk 1]
```diff
@@ -1839,12 +1842,12 @@ def to_linecollection(
...
-                raise ValueError(
-                    "Setting ``periodic_elements='split'`` is not supported when a "
-                    "projection is provided."
-                )
+        # if projection is not None:
+        #     if periodic_elements == "split":
+        #         raise ValueError(
+        #             "Setting ``periodic_elements='split'`` is not supported when a "
+        #             "projection is provided."
+        #         )
```
[the remark this answers is not in the document]
Comment (philipc2) [reply uxarray/grid/grid.py:1850]: Good catch.

File: uxarray/grid/geometry.py [hunk 2]
```diff
@@ -533,7 +534,7 @@ def _grid_to_matplotlib_polycollection(
         return PolyCollection(polygon_shells, **kwargs), []
 
 
-def _get_polygons(grid, periodic_elements, projection=None):
+def _get_polygons(grid, periodic_elements, projection=None, apply_projection=True):
```
[the remark this answers is not in the document]
Comment (philipc2) [reply uxarray/grid/geometry.py:537]: The default projection is `None`, so we don't apply one. The `apply_projection` parameter is used to set the central longitude without explcility applying a projection.

**Review by aaronzedwick — APPROVED** [review]
Looks great to me, nice work Philip!

**Review by erogluorhan — APPROVED** [review]
This looks good to me

**Pull request merged**